spring-kafka/src/main/java/org/springframework/kafka/core/KafkaAdmin.java

```diff
@@ -254,8 +257,13 @@ private void addTopics(AdminClient adminClient, List<NewTopic> topicsToAdd) {
 			throw new KafkaException("Timed out waiting for create topics results", e);
 		}
 		catch (ExecutionException e) {
-			logger.error("Failed to create topics", e.getCause());
-			throw new KafkaException("Failed to create topics", e.getCause()); // NOSONAR
+			if (e.getCause() instanceof TopicExistsException) { // Possible race with another app instance
+				logger.debug("Failed to create topics", e.getCause());
+			}
+			else {
+				logger.error("Failed to create topics", e.getCause());
+				throw new KafkaException("Failed to create topics", e.getCause()); // NOSONAR
```

Comment:
I think Sonar is not going to like a repeated String.
Let's consider to extract it as a variable!
As well as for next one about partitions.

Reply:
I think the threshold is 4 😄 

Will fix if he complains.